sunshine/platform/linux/wlgrab.cpp

```diff
@@ -154,6 +154,7 @@ class wlr_ram_t : public wlr_t {
     }
 
     auto current_frame = dmabuf.current_frame;
+    current_frame->sd.obj_count = current_frame->obj_count;
```

Comment:
Please remove `obj_count` from `frame_t` and fill `sd.obj_count` directly please.

Reply:
But there's this:
https://github.com/loki-47-6F-64/sunshine/blob/3d179a869ab32c5796e9c7fac8b9e0879974e560/sunshine/platform/linux/wlgrab.cpp#L266-L267

Also you don't really *need* the obj_count for import if you always fill the `fds` array in order - which you should do, and you will always receive it as such from external apis (fb, wlr-export-dmabuf, ...). `plane_indices` can also be removed then.

Reply:
In wayland.cpp line 174:
```
void dmabuf_t::object(
  zwlr_export_dmabuf_frame_v1 *frame,
  std::uint32_t index,
  std::int32_t fd,
  std::uint32_t size,
  std::uint32_t offset,
  std::uint32_t stride,
  std::uint32_t plane_index) {
  auto next_frame = get_next_frame();

  next_frame->sd.fds[index]           = fd;
  next_frame->sd.pitches[index]       = stride;
  next_frame->sd.offsets[index]       = offset;
  next_frame->sd.plane_indices[index] = plane_index;
}
```
What would be the point of `plane_index`. If we removed `plane_indices`, would you suggest we change the previois code to the following code?
```
void dmabuf_t::object(
  zwlr_export_dmabuf_frame_v1 *frame,
  std::uint32_t index,
  std::int32_t fd,
  std::uint32_t size,
  std::uint32_t offset,
  std::uint32_t stride,
  std::uint32_t plane_index) {
  auto next_frame = get_next_frame();

  next_frame->sd.fds[plane_index]           = fd;
  next_frame->sd.pitches[plane_index]       = stride;
  next_frame->sd.offsets[plane_index]       = offset;
}
```

Reply:
> What would be the point of plane_index

It has to be there in the protocol, since this gets called for each plane. But the way to interpret it is exactly how you suggested.

> change the previous code to the following code

Yes, that's much better + remove `plane_indices` altogether :)

Reply:
Alright, I agree it will be better overall. I'll implement the changes then.